### PR TITLE
CNDIT-757 : Update the DI Data Definition Language Scripts 

### PR DESCRIPTION
**Pull request by alog-enquizit**
CNDIT-757: Update DI DDL scripts to include the Database creation and selectively create tables only if they are absent.

### CODE REVIEW

File: report-service/src/main/resources/db/report-service-001.sql [hunk 1]
```diff
@@ -3,64 +3,105 @@
...
-CREATE TABLE elr_raw (
-    id UNIQUEIDENTIFIER PRIMARY KEY DEFAULT NEWID(),
-    message_type NVARCHAR(255) NOT NULL,
-    payload NTEXT NOT NULL,
-    created_by NVARCHAR(255) NOT NULL,
-    updated_by NVARCHAR(255) NOT NULL,
-    created_on DATETIME NOT NULL DEFAULT getdate(),
-    updated_on DATETIME NULL
-);
+IF NOT EXISTS(SELECT *
```
Comment (alog-enquizit) [inline report-service/src/main/resources/db/report-service-001.sql:6]: Tested the scripts locally by connecting to the Data Team sandbox RDS intance.

File: report-service/src/main/resources/db/report-service-001.sql [hunk 2]
```diff
@@ -1,66 +1,92 @@
...
+              FROM sys.databases
+              WHERE name = 'NBS_DataIngest')
+    BEGIN
+        CREATE DATABASE NBS_DataIngest
+    END
+GO
+USE NBS_DataIngest
+GO
 
+IF OBJECT_ID(N'dbo.elr_raw', N'U') IS NULL
```
Comment (alog-enquizit) [inline report-service/src/main/resources/db/report-service-001.sql:15]: If table is absent then create else do nothing. Likewise for all tables.

**Review by rshanmugam-enquizit — CHANGES_REQUESTED** [review]
lgtm! can you wipe the INT1 database and run the liquibase `update` command and post the output back here? once the run is good, I'll approve this PR

**Comment by alog-enquizit** [comment]
@rshanmugam-enquizit 
Synced with @ssathiah and found out that `NBS_DataIngest` database should be present before the liquibase table schemas can be created. 
As a result, we decided to create a sql script to manually create the `NBS_DataIngest` before running the `./gradlew update` which will trigger the liquibase schema update.

`IF NOT EXISTS(SELECT *
              FROM sys.databases
              WHERE name = 'NBS_DataIngest')
    BEGIN
        CREATE DATABASE NBS_DataIngest
    END
GO
USE NBS_DataIngest
GO`

I will retract this PR as a consequence of the above observation.

**Comment by rshanmugam-enquizit** [comment]
> @rshanmugam-enquizit Synced with @ssathiah and found out that `NBS_DataIngest` database should be present before the liquibase table schemas can be created. As a result, we decided to create a sql script to manually create the `NBS_DataIngest` before running the `./gradlew update` which will trigger the liquibase schema update.
> 
> `IF NOT EXISTS(SELECT * FROM sys.databases WHERE name = 'NBS_DataIngest') BEGIN CREATE DATABASE NBS_DataIngest END GO USE NBS_DataIngest GO`
> 
> I will retract this PR as a consequence of the above observation.

Interesting. We'll have the database already and we just need to make sure the tables are created only if it doesn't exist. So, we can just do that I believe right (take down create database but not the other?)

**Comment by alog-enquizit** [comment]
Validated the scripts with @ssathiah  and @rshanmugam-enquizit in the dev and INT environments.

**Review by rshanmugam-enquizit — APPROVED** [review]
ship it!

**Pull request merged**